packages/node-red-s1seven-api/lib/utils/getters.js

```diff
@@ -0,0 +1,98 @@
+const { format } = require('node:util');
+
+const {
+  DEFAULT_API_ENVIRONMENT,
+  DEFAULT_API_VERSION,
+  URL_TO_ENV_MAP,
+  GLOBAL_MODE_KEY_PATTERN,
+  GLOBAL_ACCESS_TOKEN_KEY_PATTERN,
+  DEFAULT_API_MODE,
+  GLOBAL_COMPANY_ID_KEY_PATTERN,
+} = require('../../resources/constants');
+const { getMsg, getApiConfig } = require('./async-local-storage');
+
+// configNodeIdentifier should be either the node name and if undefined the node id
+const GLOBAL_ACCESS_TOKEN_KEY = (configNode) =>
+  format(GLOBAL_ACCESS_TOKEN_KEY_PATTERN, configNode.name || configNode.id);
+
+const GLOBAL_MODE_KEY = (configNode) =>
+  format(GLOBAL_MODE_KEY_PATTERN, configNode.name || configNode.id);
+
+const GLOBAL_COMPANY_ID_KEY = (configNode) =>
+  format(GLOBAL_COMPANY_ID_KEY_PATTERN, configNode.name || configNode.id);
+
+/**
+ * @returns {'staging' | 'production'} environment
+ */
+function getApiEnvironment() {
+  const msg = getMsg();
+  const apiConfig = getApiConfig();
+  return msg.environment || apiConfig?.environment || DEFAULT_API_ENVIRONMENT;
+}
+
+/**
+ * @returns {string} base url
+ */
+function getApiUrl() {
+  const environment = getApiEnvironment();
+  const S1SEVEN_BASE_URL = process.env.S1SEVEN_BASE_URL;
+  // const S1SEVEN_BASE_URL = asyncLocalStorage.getStore().get('S1SEVEN_BASE_URL');
+  const BASE_URL = URL_TO_ENV_MAP[environment];
+  return `${S1SEVEN_BASE_URL || BASE_URL}/api`;
+}
+
+/**
+ * @returns {number} version
+ */
+function getApiVersion() {
+  const apiConfig = getApiConfig();
+  return apiConfig.apiVersion || DEFAULT_API_VERSION;
+}
+
+/**
+ * @param {import('node-red').NodeContext['global']} globalContext
+ * @returns {'test' | 'live'} mode
+ */
+function getApiMode(globalContext) {
+  const msg = getMsg();
+  const apiConfig = getApiConfig();
+  return (
+    msg.mode ||
+    globalContext.get(GLOBAL_MODE_KEY(apiConfig)) ||
```

Comment:
When calling the `verify` endpoint, `globalContext.get(GLOBAL_MODE_KEY(apiConfig))` throws an error for me which crashes Node-Red:

```
17 Mar 09:07:26 - [red] Uncaught Exception:
17 Mar 09:07:26 - [error] Error: Invalid property expression: unexpected ' ' at position 24
    at createError (/Users/eamon/work/node-red-playground-test/node_modules/@node-red/util/lib/util.js:195:13)
    at normalisePropertyExpression (/Users/eamon/work/node-red-playground-test/node_modules/@node-red/util/lib/util.js:339:23)
    at Object.getObjectProperty (/Users/eamon/work/node-red-playground-test/node_modules/@node-red/util/lib/util.js:428:24)
    at Object.value (/Users/eamon/work/node-red-playground-test/node_modules/@node-red/runtime/lib/nodes/context/index.js:355:48)
    at getApiMode (/Users/eamon/work/node-red-monorepo/packages/node-red-s1seven-api/lib/utils/getters.js:66:19)
    at verifyCertificate._inputCallback (/Users/eamon/work/node-red-monorepo/packages/node-red-s1seven-api/lib/certificates/verify-certificate.js:29:20)
    at /Users/eamon/work/node-red-playground-test/node_modules/@node-red/runtime/lib/nodes/Node.js:210:26
    at Object.trigger (/Users/eamon/work/node-red-playground-test/node_modules/@node-red/util/lib/hooks.js:166:13)
    at verifyCertificate.Node._emitInput (/Users/eamon/work/node-red-playground-test/node_modules/@node-red/runtime/lib/nodes/Node.js:202:11)
    at verifyCertificate.Node.emit (/Users/eamon/work/node-red-playground-test/node_modules/@node-red/runtime/lib/nodes/Node.js:186:25)
```

It only seems to happen when I have this config loaded:
<img width="517" alt="image" src="https://user-images.githubusercontent.com/21305201/225849028-6c15392f-d422-415e-b0ff-235f8a9caca1.png">

When using my staging config it didn't crash. I'm guessing from the error that it is because the config node's name contains a space.

Edit: confirmed, I added a space to another config name and it also crashed.

Reply:
seems like the configNode name needs to be serialized in the functions `GLOBAL_ACCESS_TOKEN_KEY`, `GLOBAL_MODE_KEY` and `GLOBAL_COMPANY_ID_KEY`